internal/kapp/kapp.go

```diff
@@ -190,3 +209,38 @@ func (c *Client) InspectJSON(appName string) (*KappInspectOutput, error) {
 
 	return &kappOutput, nil
 }
+
+// execWithUI executes a command and captures output through the UI configuration
+func (c *Client) execWithUI(command string, args []string) error {
+	// Create command
+	cmd := exec.Command(command, args...)
```

Comment:
This should be done fundamentally different. It should use cmd.NewDefaultKappCmd() as seen in: https://github.com/carvel-dev/kapp/blob/develop/cmd/kapp/kapp.go in combination with the NewConfUI which can be found in: https://github.com/cppforlife/go-cli-ui/blob/master/ui/ui.go